fix(agents): Use session ID as context_id when calling remote A2A agents

KrshnKush opened the pull request:
**Please ensure you have read the [contribution guide](https://github.com/google/adk-python/blob/main/CONTRIBUTING.md) before creating a pull request.**

### Link to Issue or Description of Change

**1. Link to an existing issue (if applicable):**

- Closes: N/A
- Related: N/A

**2. Or, if no issue exists, describe the change:**

**Problem:**
When `RemoteA2aAgent` calls a remote agent for the first time, the `context_id` in the `A2AMessage` is set to `None`. This causes the remote agent to create an independent session context, breaking session continuity between local and remote agents. Multi-turn conversations across agent boundaries fail to maintain state because each call without an existing `context_id` starts a new context on the remote side.

**Solution:**
Use the local session ID (`ctx.session.id`) as the `context_id` fallback when no existing context is available from previous remote agent responses. This ensures:
1. Session identity is maintained across local and remote agents
2. Multi-turn conversations work correctly with remote A2A agents
3. Existing `context_id` values from previous responses are preserved for conversation continuity

### Testing Plan

**Unit Tests:**

- [x] I have added or updated unit tests for my change.
- [x] All unit tests pass locally.

Added two new test cases:
1. `test_run_async_impl_uses_session_id_when_no_context_id` - Verifies session ID is used when no existing context
2. `test_run_async_impl_preserves_existing_context_id` - Verifies existing context_id is preserved

```
$ pytest tests/unittests/agents/test_remote_a2a_agent.py::TestRemoteA2aAgentExecution -v

============================= test session starts ==============================
platform darwin -- Python 3.11.14, pytest-9.0.2, pluggy-1.6.0
collected 8 items

test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_initialization_failure PASSED [ 12%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_no_message_parts PASSED [ 25%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_successful_request PASSED [ 37%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_uses_session_id_when_no_context_id PASSED [ 50%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_preserves_existing_context_id PASSED [ 62%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_a2a_client_error PASSED [ 75%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_live_impl_not_implemented PASSED [ 87%]
test_remote_a2a_agent.py::TestRemoteA2aAgentExecution::test_run_async_impl_with_meta_provider PASSED [100%]

======================== 8 passed in 1.45s ========================
```

**Manual End-to-End (E2E) Tests:**

To manually test this change:
1. Set up a local agent that uses `RemoteA2aAgent` to communicate with a remote A2A server
2. Send multiple messages to the agent within the same session
3. Verify that the `context_id` in the A2A requests matches the session ID on the first request
4. Verify that subsequent requests use the same `context_id` for conversation continuity

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/google/adk-python/blob/main/CONTRIBUTING.md) document.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have manually tested my changes end-to-end.
- [ ] Any dependent changes have been merged and published in downstream modules.

### Additional context

**Files changed:**
- `src/google/adk/agents/remote_a2a_agent.py` - Added session ID fallback for context_id
- `tests/unittests/agents/test_remote_a2a_agent.py` - Added 2 new test cases

KrshnKush committed:
Update remote_a2a_agent.py

KrshnKush committed:
Update test_remote_a2a_agent.py

KrshnKush committed:
Update test_remote_a2a_agent.py

KrshnKush commented:
Hey @ryanaiagent 
Can you please review this?

KrshnKush commented:
Hey @ryanaiagent @DeanChensj 


**CI failure is a false positive.**

The workflow uses the regex `^from.*cli.*import.*$`, which matches any import containing `cli`, including valid ones like `from a2a.client import Client` (because `client` contains `cli`).

No code in this PR imports from the `cli` package.

KrshnKush committed:
Merge branch 'main' into feat/a2a-preserve-session-identity

ryanaiagent commented:
Hi @KrshnKush , we appreciate your patience. Can you please fix the failing unit tests.

KrshnKush committed:
Merge branch 'main' into feat/a2a-preserve-session-identity

KrshnKush committed:
refactor(agents): Use or operator for context_id fallback

Replace ternary operator with more idiomatic Python or operator for
improved conciseness and readability when setting context_id fallback
to session ID.
